src/components/AnimHandler.cpp

```diff
@@ -180,11 +180,16 @@ void AnimHandler::updateAnimations(double deltaTime)
     size_t frameNum = static_cast<size_t>(m_AnimationFrame);
 
     // Check if this changed something on our animation
-    if (m_LastProcessedFrame == static_cast<size_t>(m_AnimationFrame))
-        return; // Nothing to do here // TODO: There is, if interpolation was implemented!
+    //if (m_LastProcessedFrame == static_cast<size_t>(m_AnimationFrame))
+    //    return; // Nothing to do here // TODO: There is, if interpolation was implemented!
 
     m_LastProcessedFrame = static_cast<size_t>(m_AnimationFrame);
 
+    // frameNum contains the current frame, find out what the next frame will be and how far we're in there already
+    size_t frameNext = (frameNum + 1) % (int)(numFrames); // FIXME: What happens on non-looped animation on the last frame?
```

Comment:
Actually, we should add another function, like `truncate` or something, because floor behaves a little bit differently for negative numbers.